docs/tutorial.rst

```diff
@@ -1376,9 +1376,36 @@ motors in motion at once, including multiple fast motors and one slow motor. We
 want to wait for the fast motors to arrive, print a message, then wait for the
 slow motor to arrive, and print a second message.
 
+For Bluesky version v1.13.0 and higher, this approach is recommended:
+
 .. code-block:: python
 
     def staggered_wait(fast_motors, slow_motor):
+        # Start all the motors, fast and slow, moving at once.
+        # Put all the fast_motors in one group...
+        statuses = []
+        for motor in fast_motors:
+            status = yield from bps.abs_set(motor, 5)
```

Comment:
A maybe issue that this example brought to mind: we are also adding these statuses to the default group!

If we have plans that _only_ use this aproach, we will ever clear out those cached statuses or will we have an unbounded list that will (eventually) oom us?

Reply:
@tacaswell I don't think I quite understand what the issue is. Are the statuses not added to their own group labeled with a hash?

Reply:
```
 status = yield from bps.abs_set(motor, 5)
```

This will add to the `None` group (as we can not predict how it will be used in the future!)

The 

```
yield from bsp.wait(status)
```

will add it to the (hopefully short lived) group.

The group created via the `wait` will be dropped, but I don't think it will get removed from the `None` group.

Reply:
Why do we maintain this `None` group? If group=None is passed couldn't we simply not associate the status with any group?

Reply:
It is an early design choice to let

```python
yield from bps.abs_set(mtr1, 1)
yield from bps.bas_set(mtr2, 2)
yield from bps.wait()
```

"just work" with an implicit group.  Dropping that would likely break a bunch of stuff out in the wild so we can not just unilaterally change that.

Maybe we want to add a special group for "do not track me!",  but that would defeat a bunch of the benefits of adding the ability to pass status objects (the reduced boiler plate).  I would also have concerns about that because in addition to being able to wait on the status object the RE does a bunch of book keeping to be able to raise-on-motion-failure and having a loose status object is a bit of an anathema to the design (the first job of the RE is to track when things you asked hardware to do are done! as I said else where if there is something you will never care when it is actually done, then it should return an already completed Status object as that is the right way to nope out of the RE doing its job).

This also makes we wonder if when combined with move on and calling `wait` with different sets of status objects is going to result in status object being in multiple groups?  Something like:

```
st_list = [...]
while len(st_list):
   st_list = [st for st in st_list if not st.done]
   yield from bps.wait(st_list, timeout=.5, move_on=True)
```

so even if we did drop the implicit `None` group, we could still drive ourselves to an unbounded cache (I think).

Reply:
Would it then be possible to check if the statuses exist in the `None` group, pop them, and put them in a dedicated group?

Reply:
I think possible, but would we want to pop them out forever or put them back if unexpired?

```python
st_list = [...]
yield from bps.wait(st_list, timeout=.5, move_on=True)
yield from bps.wait()  # <- this should wait on anything not yet done!
```

Reply:
I am able to pop the status objects from `self._status_objs[None]' and put them in a new set under a unique group name. The issue is I can't figure out how I can retrieve their futures from `self._groups[None]` to put into a new group of futures.

```python
        if isinstance(group, collections.abc.Iterable) and not isinstance(group, str):
            status_objs = group
            group = str(hash(tuple(sorted(id(s) for s in group))))
            for status in status_objs:
                # self._add_status_to_group(obj=None, status_object=status, group=group, action="wait")
                if status in self._status_objs[None]:
                    self._status_objs[None].remove(status)
                    self._status_objs[group].add(status)
```